### PR TITLE
remove extra decimal places from energies and stopping powers

### PR DESCRIPTION
**Pull request by piotrpolec**
The precision is the same as in dedx

CSDA ranges will be done in a separate commit

### CODE REVIEW

**Comment by grzanka** [comment]
@piotrpolec please resolve conflicts

**Comment by piotrpolec** [comment]
@grzanka done

**Comment by grzanka** [comment]
As suggested in https://github.com/APTG/dedx_web/issues/137 I'd like to have 3 digits after the dot in displayed values.
The precision should be much higher in the exported CSV files (10 digits).

**Review by grzanka — CHANGES_REQUESTED** [review]
See comments

**Comment by grzanka** [comment]
> 3 digits don't make sense with default energies: first two points are 0.001 and 0.0015 which can have vastly different stopping power (133 and 163 for hydrogen in water) but will be shown as the same point

Good point:

To be more specific:
 - for placing points on the plot no rounding should be applied
 - we will see how users react, but most probably table of numbers presented in PLOT tab should disappear, but the "Download CSV" should stay
 - for table on the "PLOT" tab we should use for now for presentation on the page:
   - 4 digits after dot for energy 
   - 3 digits after dot for stopping power
   - for dump to CSV for all numbers 10 digits after dot should be used
 - for table on the "DATA" tab we should use for now for presentation on the page:
   - 4 digits after dot for energy 
   - 3 digits after dot for stopping power and range (I assume that later https://github.com/APTG/dedx_web/issues/162 will be fixed)
   - for dump to CSV for all numbers 10 digits after dot should be used

**Review by Mexolius — APPROVED** [review]
Most comments were adressed and the changes aren't going directly to master - we can see how it behaves when merged with new UI and fix them before the general master merge

File: src/Backend/WASMWrapper.js [hunk 1]
```diff
@@ -198,7 +198,7 @@ export default class WASMWrapper {
             [...ids, _energies.byteOffset, _stoppingPowers.byteOffset]
         )
 
-        const energies = !err ? Array.from(_energies).map(energy => energy.toFixed(6)) : [0]
+        const energies = !err ? Array.from(_energies).map(energy => energy.toFixed(6).replace(/(\d)0+$/gm, '$1')) : [0]
```
Comment (grzanka) [inline src/Backend/WASMWrapper.js:201]: Regular expressions are not self-explanatory. The purpose of using them here is not clear to code reader.
Add an appropriate comment explaining:
 - what is the reason behind using such code
 - what this code is doing

**Comment by grzanka** [comment]
What about CSDA ranges ?
image

**Comment by piotrpolec** [comment]
csda ranges are going to be dealt with in a separate PR since the library used for converting doesn't support smaller units than cm and we cant just round it to 3 numbers after the decimal

**Review by grzanka — APPROVED** [review]
LGTM

**Pull request merged**